src/function.ts

```diff
@@ -0,0 +1,166 @@
+import { Config, Connection } from "./connection";
+
+export interface DeployQueryWorkerParameters {
+  type: string;
+  name: string;
+  queryWorkerName: string;
+  environment: string;
+}
+
+export interface DeployStreamAdhocQueryParameters {
+  type: string;
+  name: string;
+  streamWorkerName: string;
+  environment: string;
+}
+
+export interface DeployStreamPublisherParameters {
+  type: string;
+  name: string;
+  streamWorkerName: string;
+  streamName: string;
+  environment: string;
+}
```

Comment:
if we are using an interface then i think here we can extend it as `type, name, environment` are common.